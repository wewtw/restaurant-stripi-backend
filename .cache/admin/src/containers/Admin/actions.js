/*
 *
 * Admin actions
 *
 */

import {
  EMIT_EVENT,
  GET_INIT_DATA,
  GET_INIT_DATA_SUCCEEDED,
  GET_SECURED_DATA,
  GET_SECURED_DATA_SUCCEEDED,
  HIDE_LEFT_MENU,
  HIDE_LOGOUT,
  SET_APP_ERROR,
  SET_APP_SECURED,
  SHOW_LEFT_MENU,
  SHOW_LOGOUT,
  UNSET_APP_SECURED,
} from './constants';

export function emitEvent(event, properties) {
  return {
    type: EMIT_EVENT,
    event,
    properties,
  };
}

export function getInitData() {
  return {
    type: GET_INIT_DATA,
  };
}

export function getInitDataSucceeded(data) {
  return {
    type: GET_INIT_DATA_SUCCEEDED,
    data,
  };
}

export function getSecuredData() {
  return {
    type: GET_SECURED_DATA,
  };
}

export function getSecuredDataSucceeded(data) {
  return {
    type: GET_SECURED_DATA_SUCCEEDED,
    data,
  };
}

export function hideLeftMenu() {
  return {
    type: HIDE_LEFT_MENU,
  };
}

export function hideLogout() {
  return {
    type: HIDE_LOGOUT,
  };
}

export function setAppError() {
  return {
    type: SET_APP_ERROR,
  };
}

export function setAppSecured() {
  return {
    type: SET_APP_SECURED,
  };
}

export function showLeftMenu() {
  return {
    type: SHOW_LEFT_MENU,
  };
}

export function showLogout() {
  return {
    type: SHOW_LOGOUT,
  };
}

export function unsetAppSecured() {
  return {
    type: UNSET_APP_SECURED,
  };
}
