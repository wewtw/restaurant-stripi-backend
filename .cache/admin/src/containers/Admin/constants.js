/*
 *
 * Admin constants
 *
 */

export const EMIT_EVENT = 'app/Admin/EMIT_EVENT';
export const GET_INIT_DATA = 'StrapiAdmin/Admin/GET_INIT_DATA';
export const GET_INIT_DATA_SUCCEEDED =
  'StrapiAdmin/Admin/GET_INIT_DATA_SUCCEEDED';
export const GET_SECURED_DATA = 'StrapiAdmin/Admin/GET_SECURED_DATA';
export const GET_SECURED_DATA_SUCCEEDED =
  'StrapiAdmin/Admin/GET_SECURED_DATA_SUCCEEDED';
export const HIDE_LEFT_MENU = 'StrapiAdmin/Admin/HIDE_LEFT_MENU';
export const HIDE_LOGOUT = 'StrapiAdmin/Admin/HIDE_LOGOUT';
export const SET_APP_ERROR = 'StrapiAdmin/Admin/SET_APP_ERROR';
export const SET_APP_SECURED = 'StrapiAdmin/Admin/SET_APP_SECURED';
export const SHOW_LEFT_MENU = 'StrapiAdmin/Admin/SHOW_LEFT_MENU';
export const SHOW_LOGOUT = 'StrapiAdmin/Admin/SHOW_LOGOUT';
export const UNSET_APP_SECURED = 'StrapiAdmin/Admin/UNSET_APP_SECURED';
