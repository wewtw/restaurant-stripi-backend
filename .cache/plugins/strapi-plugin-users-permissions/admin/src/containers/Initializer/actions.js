/*
 *
 * Initializer actions
 *
 */

import {
  INITIALIZE,
  INITIALIZE_SUCCEEDED,
  UPDATE_HAS_ADMIN,
} from './constants';

export function initialize() {
  return {
    type: INITIALIZE,
  };
}

export function initializeSucceeded(data) {
  return {
    type: INITIALIZE_SUCCEEDED,
    data,
  };
}

export function updateHasAdmin(value) {
  return {
    type: UPDATE_HAS_ADMIN,
    value,
  };
}
