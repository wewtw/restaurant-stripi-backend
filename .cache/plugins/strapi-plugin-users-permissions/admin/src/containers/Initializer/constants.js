/*
 *
 * Initializer constants
 *
 */

export const INITIALIZE = 'UsersPermissions/Initializer/INITIALIZE';
export const INITIALIZE_SUCCEEDED =
  'UsersPermissions/Initializer/INITIALIZE_SUCCEEDED';
export const UPDATE_HAS_ADMIN = 'UsersPermissions/Initializer/UPDATE_HAS_ADMIN';
