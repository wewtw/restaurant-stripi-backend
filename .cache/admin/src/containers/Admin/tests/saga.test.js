/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */
/* eslint-disable redux-saga/no-unhandled-errors */
import { all, fork, put, takeLatest } from 'redux-saga/effects';
import defaultSaga, { emitter, getData, getSecuredData } from '../saga';

import { getInitDataSucceeded } from '../actions';
import { EMIT_EVENT, GET_INIT_DATA, GET_SECURED_DATA } from '../constants';

describe('getData Saga', () => {
  let getDataGenerator;

  beforeEach(() => {
    getDataGenerator = getData();
    const response = [
      { uuid: 'uuid' },
      { strapiVersion: 'beta' },
      { autoReload: true, currentEnvironment: 'test' },
      { layout: {} },
    ];
    const callDescriptor = getDataGenerator.next(response).value;

    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the getInitDataSucceeded action if it requests the data successfully', () => {
    const response = [
      { uuid: 'uuid' },
      { strapiVersion: 'beta' },
      { autoReload: true, currentEnvironment: 'test' },
      { layout: {} },
    ];
    const putDescriptor = getDataGenerator.next(response).value;
    const [
      { uuid },
      { strapiVersion },
      { autoReload, currentEnvironment },
      { layout },
    ] = response;

    expect(putDescriptor).toEqual(
      put(
        getInitDataSucceeded({
          autoReload,
          uuid,
          strapiVersion,
          currentEnvironment,
          layout,
        })
      )
    );
  });

  it('should call the strapi.notification action if the response errors', () => {
    const response = new Error('Some error');
    getDataGenerator.throw(response).value;

    expect(strapi.notification.error).toHaveBeenCalled();
  });
});

describe('defaultSaga Saga', () => {
  const defaultSagaSaga = defaultSaga();

  it('should start task to watch for GET_INIT_DATA and GET_SECURED_DATA actions', () => {
    const forkDescriptor = defaultSagaSaga.next().value;

    expect(forkDescriptor).toEqual(
      all([
        fork(takeLatest, EMIT_EVENT, emitter),
        fork(takeLatest, GET_INIT_DATA, getData),
        fork(takeLatest, GET_SECURED_DATA, getSecuredData),
      ])
    );
  });
});
