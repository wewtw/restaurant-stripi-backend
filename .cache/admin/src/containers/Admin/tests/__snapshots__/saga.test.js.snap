// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData Saga should call the strapi.notification action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "ALL": Array [
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "/admin/gaConfig",
          Object {
            "method": "GET",
          },
        ],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "/admin/strapiVersion",
          Object {
            "method": "GET",
          },
        ],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "/admin/currentEnvironment",
          Object {
            "method": "GET",
          },
        ],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "/admin/layout",
          Object {
            "method": "GET",
          },
        ],
        "context": null,
        "fn": [Function],
      },
    },
  ],
}
`;

exports[`getData Saga should dispatch the getInitDataSucceeded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "ALL": Array [
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "/admin/gaConfig",
          Object {
            "method": "GET",
          },
        ],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "/admin/strapiVersion",
          Object {
            "method": "GET",
          },
        ],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "/admin/currentEnvironment",
          Object {
            "method": "GET",
          },
        ],
        "context": null,
        "fn": [Function],
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          "/admin/layout",
          Object {
            "method": "GET",
          },
        ],
        "context": null,
        "fn": [Function],
      },
    },
  ],
}
`;
